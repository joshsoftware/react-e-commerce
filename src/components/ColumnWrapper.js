import React from 'react';
import { Col } from 'reactstrap';
import PropTypes from 'prop-types';

const ColumnWrapper = ({ data, xs, sm, md, lg, xl, widths }) => {
  return (
    <Col xs={xs} sm={sm} md={md} lg={lg} xl={xl} widths={widths}>
      {data}
    </Col>
  );
};

export default ColumnWrapper;

const stringOrNumberProp = PropTypes.oneOfType([PropTypes.number, PropTypes.string]);
const columnProps = PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.number,
  PropTypes.bool,
  PropTypes.shape({
    size: PropTypes.oneOfType([PropTypes.bool, PropTypes.number, PropTypes.string]),
    order: stringOrNumberProp,
    offset: stringOrNumberProp
  })
]);

ColumnWrapper.propTypes = {
  data: PropTypes.oneOfType([PropTypes.element, PropTypes.array, PropTypes.object]),
  xs: columnProps,
  sm: columnProps,
  md: columnProps,
  lg: columnProps,
  xl: columnProps,
  widths: PropTypes.array
};
