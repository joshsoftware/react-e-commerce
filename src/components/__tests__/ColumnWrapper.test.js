import React from 'react';
import ColumnWrapper from '../ColumnWrapper';
import { render } from '@testing-library/react';

describe('ColumnWrapper Component', () => {
  it('Must return column', () => {
    const { asFragment } = render(<ColumnWrapper />);

    expect(asFragment()).toMatchSnapshot();
  });
});
