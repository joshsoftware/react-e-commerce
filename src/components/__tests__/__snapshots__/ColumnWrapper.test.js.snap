// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnWrapper Component Must return column 1`] = `
<DocumentFragment>
  <div
    class="col"
  />
</DocumentFragment>
`;
